dali/python/nvidia/dali/plugin/pytorch/__init__.py

```diff
@@ -268,23 +269,36 @@ def __next__(self) -> List[Dict[str, torch.Tensor]]:
                     category_device[category] = torch_cpu_device
 
             pyt_tensors = dict()
-            for category in self._output_categories:
-                pyt_tensors[category] = torch.empty(
-                    category_shapes[category],
-                    dtype=category_torch_type[category],
-                    device=category_device[category],
-                )
 
-            data_batches[i] = pyt_tensors
+            copy = not is_exec_dynamic
+            if copy:
+                # Copy data from DALI Tensors to torch tensors
+                for category, tensor in category_tensors.items():
+                    pyt_tensor = torch.empty(
+                        category_shapes[category],
+                        dtype=category_torch_type[category],
+                        device=category_device[category],
+                    )
+                    pyt_tensors[category] = pyt_tensor
 
-            # Copy data from DALI Tensors to torch tensors
-            for category, tensor in category_tensors.items():
-                if isinstance(tensor, (TensorGPU, TensorListGPU)):
-                    # Using same cuda_stream used by torch.zeros to set the memory
-                    stream = torch.cuda.current_stream(device=pyt_tensors[category].device)
-                    feed_ndarray(tensor, pyt_tensors[category], cuda_stream=stream)
-                else:
-                    feed_ndarray(tensor, pyt_tensors[category])
+                    if isinstance(tensor, TensorGPU):
```

Comment:
in the previous code we were also matching TensorListGPU. Can you provide some background for this change?

Reply:
It was dead code. We replace TensorList with a tensor in line 254:
```Python
category_tensors[category] = out.as_tensor()
```